test/e2e/common/node/ephemeral_containers.go

```diff
@@ -80,4 +86,88 @@ var _ = SIGDescribe("Ephemeral Containers [NodeConformance]", func() {
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "test-container-1",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"/bin/sleep"},
+						Args:    []string{"10000"},
+					},
+				},
+			},
+		})
+
+		ginkgo.By("adding an ephemeral container")
+		ecName := "debugger"
+		ec := &v1.EphemeralContainer{
+			EphemeralContainerCommon: v1.EphemeralContainerCommon{
+				Name:    ecName,
+				Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+				Command: e2epod.GenerateScriptCmd("while true; do echo polo; sleep 2; done"),
+				Stdin:   true,
+				TTY:     true,
+			},
+		}
+		err := podClient.AddEphemeralContainerSync(ctx, pod, ec, time.Minute)
+		framework.ExpectNoError(err, "Failed to patch ephemeral containers in pod %q", e2epod.FormatPod(pod))
+
+		ginkgo.By("checking pod container endpoints")
+		// Can't use anything depending on kubectl here because it's not available in the node test environment
+		output := e2epod.ExecCommandInContainer(f, pod.Name, ecName, "/bin/echo", "marco")
+		gomega.Expect(output).To(gomega.ContainSubstring("marco"))
+		log, err := e2epod.GetPodLogs(ctx, f.ClientSet, pod.Namespace, pod.Name, ecName)
+		framework.ExpectNoError(err, "Failed to get logs for pod %q ephemeral container %q", e2epod.FormatPod(pod), ecName)
+		gomega.Expect(log).To(gomega.ContainSubstring("polo"))
+
+		ginkgo.By(fmt.Sprintf("checking pod %q has only one ephemeralcontainer", pod.Name))
+		podResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+		unstruct, err := f.DynamicClient.Resource(podResource).Namespace(f.Namespace.Name).Get(ctx, "ephemeral-containers-target-pod", metav1.GetOptions{}, "ephemeralcontainers")
+		framework.ExpectNoError(err, "can't get ephermalcontainers: %#v", err)
+		verifyPod, err := unstructuredToPod(unstruct)
+		framework.ExpectNoError(err, "Getting the pod %q ephemeralcontainers", verifyPod.Name)
+		gomega.Expect(verifyPod.Spec.EphemeralContainers).To(gomega.HaveLen(1), "checking ephemeralContainer count")
+
+		ginkgo.By(fmt.Sprintf("adding another ephemeralcontainer to pod %q", pod.Name))
+		var podToUpdate *v1.Pod
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			podToUpdate, err = podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Unable to retrieve pod %s", pod.Name)
+
+			podToUpdate.Spec.EphemeralContainers = append(podToUpdate.Spec.EphemeralContainers, v1.EphemeralContainer{
+				EphemeralContainerCommon: v1.EphemeralContainerCommon{
+					Name:                     "debugger2",
+					Image:                    imageutils.GetE2EImage(imageutils.Agnhost),
+					ImagePullPolicy:          "IfNotPresent",
+					TerminationMessagePolicy: "File",
+				},
+			})
+			_, err = podClient.UpdateEphemeralContainers(context.TODO(), pod.Name, podToUpdate, metav1.UpdateOptions{})
+			return err
+		})
+		framework.ExpectNoError(err, "Failed to update status. %v", err)
```

Comment:
no need to have `err` in both - first parameter and string substitute. It will be written twice.

Also typically context message is about something was hepenning, not starting with "Failed". In this case it will be "updating ephemeral container" or sometjing like this

Reply:
Thanks for catching the issue with the message context. It was from another test I've done for status endpoints. Fixed.